### ISSUE TITLE
customLinkPrefixes isn't plumbed throughout library

### ISSUE THREAD

**Issue by mikecodega** [singerdmx/flutter-quill#1776]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Flutter Quill version

9.3.1

### Steps to reproduce

When customLinkPrefixes are set in QuillEditorConfigurations...

- They aren't honored in the _LinkDialog from the toolbar.
  - this can be reproduced by (1) setting them in the config, (2) trying to use them in the dialog.
- They aren't honored during insert by AutoFormatMultipleLinksRule
  - this can be reproduced by trying to insert text when using the rule.

### Expected results

Expect that custom Link Prefixes are honored for the above fields.

I would be happy to help with writing a PR for the fix, but I think the library maintainers should suggest a path for the fix. The library mostly uses Regex to detect links, but the customLinkPrefixes is just a list of strings of the prefixes. Additionally, some parts of the library that interact with links aren't configured via the QuillEditorConfigurations. I'm not experienced with the library's architecture enough to make them without input. 

### Actual results

Custom Link Prefixes are ignored for the above fields.

### Code sample

n/a this can be validated using the repo's provided examples, with this one config added.

**Comment by CatHood0** [comment]
Does this isse still persists? I will close it. Please, reopen if it does.

**Comment by EchoEllet** [comment]
@mikecodega Can you confirm if this still an issue?

**Comment by mikecodega** [comment]
Hey, this issue doesn't seem to be resolved. What changes were made? 

It looks like it's still not possible to use the built-in link insertion modal from the toolbar to insert a link with a custom prefix. 

For example, I tried "scheme://" as a prefix, and then adding a link such as "scheme://XXXX", and the only thing it'd accept is "http://XXXX" or "https://XXXX"

**Comment by orevial** [comment]
I can confirm that `customLinkPrefixes` is not honored, it's not passed at all in the actual link detection rule.

I think the fix might not be too heavy, we should pass this parameter to `AutoFormatMultipleLinksRule` and generate more link patterns for link detection, because currently quill editor enforces only one pattern: 

```dart
static const _oneLineLinkPattern =
      r'^https?:\/\/[\w\-]+(\.[\w\-]+)*(:\d+)?([\/\?#].*)?$';
  static const _detectLinkPattern =
      r'https?:\/\/[\w\-]+(\.[\w\-]+)*(:\d+)?([\/\?#][^\s]*)?';
```

Something like this might work (it's only pseudo-code though): 

```dart
customLinkPrefixes.forEach((customPrefix) {
  final pattern = '^${customPrefix}?:\/\/[\w\-]+(\.[\w\-]+)*(:\d+)?([\/\?#].*)?$';

  // Check if the custom pattern matches 
  if(matches)
     // proceed with match on next steps
  else
     // continue in for loop
});
```

**Comment by LP0815** [comment]
It would really be great if someone could fix this. Even simple email links like "mailto:hello@world.com" are not accepted in the input dialog after adding `customLinkPrefixes: const ['mailto:', 'tel:']` in the "QuillEditorConfigurations"

**Comment by EchoEllet** [comment]
@CatHood0

I will take over, I noticed you've been trying to do too much recently.

**Comment by CatHood0** [comment]
> I will take over, I noticed you've been trying to do too much recently.

Thank you so much 🙇🏻‍♂️!

**Comment by EchoEllet** [comment]
> For example, I tried "scheme://" as a prefix, and then adding a link such as "scheme://XXXX", and the only thing it'd accept is "http://XXXX" or "https://XXXX"

Yes, it isn't clear; see the [related lines](https://github.com/singerdmx/flutter-quill/blob/f376f72545adce8a41cae6d963ea010ced14ca25/lib/src/editor/widgets/text/text_line.dart#L674-L689).
